Fixes #30 UI addition for signup and login modal

vanamraghu opened the pull request:
1. KlinicAuthform that handles both login & signUp modal
2. KlinicAuthContainer that has tabs to switch b/w login and signup and render the login/signup form
3. Added clsx in package json and installed to handle multiple classes

## What type of PR is this? (check all applicable)

- [ ] Feature

## Description

Addition of login button in header so that when clicked it should display the modal and when close button is clicked it closes the modal 
Addition of Signup and Login modal to handle sign up and login process,
Form is added that can be reused that has Email and password


## Related Tickets & Documents

- Related Issue #30 
- Closes #30 

## QA Instructions, Screenshots, Recordings

1. Testing is done by adding Header that would render the login button

![CleanShot 2024-10-16 at 11 32 10](https://github.com/user-attachments/assets/93ead386-a3ec-4785-86df-043cd0a07377)

2. Clicking on Login would display below form

![CleanShot 2024-10-16 at 11 32 48](https://github.com/user-attachments/assets/b9e36b24-389b-49c1-829a-97bf3e85314a)

3. By default signup is highlighted, when login is clicked it would show login form

![CleanShot 2024-10-16 at 11 33 31](https://github.com/user-attachments/assets/cad3454e-4046-4da6-a919-834e5704a93f)

4. Closing the button would close the modal or clicking anywhere outside the modal would close the modal

vanamraghu committed:
Fixes #30 UI addition for signup and login modal
1. KlinicAuthform that handles both login & signUp modal
2. KlinicAuthContainer that has tabs to switch b/w login and signup and render the login/signup form
3. Added clsx in package json and installed to handle multiple classes

rt4914 reviewed (changes requested):
@vanamraghu I have created this video which should help you understand how we should think about creating scalable solutions. 

https://www.loom.com/share/0ee4064f30ac49e3afe1cc2ae9175c89?sid=1f947cac-e84d-435c-a05c-bc2bd456883d

vanamraghu committed:
Merge branch 'main' of https://github.com/rt4914/KlinicCon-Frontend into 30-feature-request-ui-add-loginsignup-modal

vanamraghu committed:
Fixes #30 UI addition for signup and login modal
1. KlinicAuthForm renamed to AuthForm that displays the form for login and sign up
2. Modal component created to display the login and sign up form and also state is handled to display modal when close button is clicked or anywhere in the page
3. KlinicAuthContainer is removed and renamed to Modal component
4. Button component modified so that first logo can be displayed following the button text (earlier it was text followed by children) now children is displayed followed by button text
5. Header component is created to display the LogIn button so that when clicked modal and related forms is displayed

vanamraghu committed:
Merge branch 'main' into 30-feature-request-ui-add-loginsignup-modal

vanamraghu committed:
Fixes #30 UI addition for signup and login modal
1. Removed clsx and making use of classNames to handle display of conditionals for tailwind

vanamraghu committed:
Merge remote-tracking branch 'origin/30-feature-request-ui-add-loginsignup-modal' into 30-feature-request-ui-add-loginsignup-modal

# Conflicts:
#	package-lock.json

rt4914 commented:
Unassigned myself as this PR is not ready for the review.
